commit 81156928f8fe31621e467490b9d441c0285998c3
Author: Pavel Roskin <proski@gnu.org>
Date:   Sat Jan 17 13:33:03 2009 -0500

    dell_rbu: use scnprintf() instead of less secure sprintf()
    
    Reading 0 bytes from /sys/devices/platform/dell_rbu/image_type or
    /sys/devices/platform/dell_rbu/packet_size by an ordinary user causes an
    oops.
    
    Signed-off-by: Pavel Roskin <proski@gnu.org>
    Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>

Adjusted to apply to Debian's 2.6.26 by dann frazier <dannf@debian.org>

diff -urpN linux-source-2.6.26.orig/drivers/firmware/dell_rbu.c linux-source-2.6.26/drivers/firmware/dell_rbu.c
--- linux-source-2.6.26.orig/drivers/firmware/dell_rbu.c	2008-07-13 15:51:29.000000000 -0600
+++ linux-source-2.6.26/drivers/firmware/dell_rbu.c	2009-02-09 21:44:12.000000000 -0700
@@ -598,7 +598,7 @@ static ssize_t read_rbu_image_type(struc
 {
 	int size = 0;
 	if (!pos)
-		size = sprintf(buffer, "%s\n", image_type);
+		size = scnprintf(buffer, count, "%s\n", image_type);
 	return size;
 }
 
@@ -670,7 +670,7 @@ static ssize_t read_rbu_packet_size(stru
 	int size = 0;
 	if (!pos) {
 		spin_lock(&rbu_data.lock);
-		size = sprintf(buffer, "%lu\n", rbu_data.packetsize);
+		size = scnprintf(buffer, count, "%lu\n", rbu_data.packetsize);
 		spin_unlock(&rbu_data.lock);
 	}
 	return size;
